apps/heft/src/plugins/TypeScriptPlugin/Eslint.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import * as crypto from 'crypto';
+import * as TEslint from 'eslint';
+
+import { LinterBase, ILinterBaseOptions } from './LinterBase';
+import { IExtendedSourceFile } from './internalTypings/TypeScriptInternals';
+import { IColorableSequence, Colors } from '@rushstack/node-core-library';
+
+interface IEslintOptions extends ILinterBaseOptions {
+  eslintPackagePath: string;
+}
+
+export class Eslint extends LinterBase<TEslint.ESLint.LintResult> {
+  private readonly _eslintPackage: typeof TEslint;
+
+  private _eslintCli: TEslint.CLIEngine;
+  private _eslint: TEslint.ESLint;
+  private _eslintBaseConfiguration: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+  private _lintResult: TEslint.ESLint.LintResult[];
+
+  public constructor(options: IEslintOptions) {
+    super('eslint', options);
+
+    this._eslintPackage = require(options.eslintPackagePath);
+  }
+
+  public printVersionHeader(): void {
+    this._terminal.writeLine(`Using ESLint version ${this._eslintPackage.Linter.version}`);
+  }
+
+  public reportFailures(): void {
+    const eslintFailureLogMessages: (string | IColorableSequence)[][] = [];
+
+    for (const eslintFileResult of this._lintResult) {
+      const buildFolderRelativeFilePath: string = path.relative(
+        this._buildFolderPath,
+        eslintFileResult.filePath
+      );
+      for (const message of eslintFileResult.messages) {
+        // https://eslint.org/docs/developer-guide/nodejs-api#◆-lintmessage-type
```

Comment:
Better to make a `const enum ESlintSeverity` and document that, rather than using the integers directly.